### ISSUE TITLE
val declarations inside generators need to be assigned only once

### ISSUE THREAD

**Issue by Contentmaudlin** [shux-lang/shux#134]
Legal:
```
gn foo(int x, int y) int {
   int a;
   a = x + y;
   a   
}
```

Illegal:
```
gn(int x) { 
    int a = x;
    a = 2;
    a
}
```

- Formals can't be assigned. 
- vars can be assigned.
- vals can be assigned **only once**. This includes initialization.

**Comment by Contentmaudlin** [comment]
@cerrno Test cases needed for this.

**Comment by Contentmaudlin** [comment]
This should also be implemented within the following fix: #137
